app/controllers/api/v1/application_users_controller.rb

```diff
@@ -92,6 +92,34 @@ def index
   end
 
   ###############################################################
+  # username
+  ###############################################################
+
+  api :GET, '/application_users/username/:username',
+      'Gets a single User with the specified username.'
+  description <<-EOS
+      All users of OpenStax have an associated User object.  This endpoint
+      allows querying additional data on a user when all that is known is the
+      User's username.
+
+      Admins (for Accounts only) are identified by the is_administrator boolean.
+      Some additional user information can be found in associations, such as
+      email addresses in ContactInfos and the password hash in Identity.
+
+      Users have the following String attributes:
+      username, first_name, last_name, full_name, title
+
+    #{json_schema(Api::V1::UserSearchRepresenter, include: :readable)}
+  EOS
+  def username
+    OSU::AccessPolicy.require_action_allowed!(:search, current_api_user, ApplicationUser)
+    username = params[:username]
+    # N.B. references{:user} must be added to the query when/if upgraded to Rails4
+    respond_with ApplicationUser.includes{user}.where{user.username.eq username},
```

Comment:
I would do a .includes(:user).joins(:user) to force it to be an inner join (brackets work too).

Not a huge problem here I don't think, but Rails sometimes feels like splitting includes into multiple queries, which would cause an error, since the where clause references the users table.